### PR TITLE
feat(commands): create create-parser command

### PR DESCRIPTION
**Pull request by AllanMichay**
## What it does

Add a command `yarn create-parser` to easily create a parser using a skeleton

### CODE REVIEW

File: create-parser.sh [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+beautifiedName=$(echo "${nameWithSpace}" | awk '{for(i=1;i<=NF;i++)sub(/./,toupper(substr($i,1,1)),$i)}1')
+
+# Copy in new parser to keep parser-skeleton as is
+cp -R parser-skeleton "./parsers/$name";
+
+# Replace the content. Using .bak so that we are cross platform https://stackoverflow.com/a/22084103
+sed -i.bak 's;../parsers/global-libs;../global-libs;g' "parsers/$name/parser-skeleton.parser.ts";
+sed -i.bak 's/parser-skeleton/'"$name"'/g' "parsers/$name/parser-skeleton.spec.ts";
+sed -i.bak 's/parser-skeleton/'"$name"'/g' "parsers/$name/README.md";
+sed -i.baku 's/parser-name-skeleton/'"$beautifiedName"'/g' "parsers/$name/README.md";
```
Comment (AMoreaux) [inline create-parser.sh:21]: `.baku`? It's normal?

File: parser-skeleton/parser-skeleton.spec.ts [hunk 2]
```diff
@@ -0,0 +1,17 @@
+// The token type. It contains shared informations and methods.
```
Comment (AMoreaux) [inline parser-skeleton/parser-skeleton.spec.ts:1]: Add more example of tests to encourage tests writing

File: parser-skeleton/README.md [hunk 3]
```diff
@@ -0,0 +1,72 @@
+# parser-name-skeleton
```
Comment (AMoreaux) [inline parser-skeleton/README.md:1]: In `.sh` script. Which line renames this part?

**Review by AMoreaux — CHANGES_REQUESTED** [review]

**Review by AMoreaux — DISMISSED** [review]

**Review by AMoreaux — APPROVED** [review]

File: README.md [hunk 4]
```diff
@@ -1,17 +1,14 @@
...
-
-
-
 
+<!-- [![CI Actions Status](https://github.com/Specifyapp/parsers/workflows/CI/badge.svg)](https://github.com/Specifyapp/parsers/actions) -->
 
 # Specify Parsers
 
-Specify is a central platform for companies who want to unify their brand identity, by connecting their design system tools. 
+Specify is a central platform for companies who want to unify their brand identity, by connecting their design system tools.
```
Comment (ChucKN0risK) [inline README.md:11]: Please replace by our current baseline: "Specify helps you unify your brand identity by collecting, storing and distributing design tokens and assets — automatically."

File: README.md [hunk 5]
```diff
@@ -25,14 +22,17 @@ By using parsers, you dictate the way you receive the data from Specify to fit y
...
 ## Creating your own parser
 
-1. Git clone this repo
-2. In the directory `parsers`, create a directory with your parser's name
-3. Create your valid parser
-4. Make a PR
+1. Fork this repo
+2. Clone the forked repo
+3. Execute `yarn`
+4. Execute `yarn create-parser [PARSER NAME]`. Parser name should be in kebab-case.
```
Comment (ChucKN0risK) [inline README.md:28]: Replace "kebab-case" by "kebabcase"

File: README.md [hunk 6]
```diff
@@ -25,14 +22,17 @@ By using parsers, you dictate the way you receive the data from Specify to fit y
...
+5. Create your valid parser
+6. Make a PR
 
 ### Creating a valid parser
 
 To be valid, a parser needs:
+
 - A `README.md` file which contains:
```
Comment (ChucKN0risK) [inline README.md:36]: Replace "which contains:" by "containing"

File: README.md [hunk 7]
```diff
@@ -44,10 +44,13 @@ To be valid, a parser needs:
 ## Usables libraries
 
 For now, our parsers only use the following libraries:
+
 - [tinycolor2](https://github.com/bgrins/TinyColor)
 - [lodash](https://github.com/lodash/lodash)
+- [svgo](https://github.com/svg/svgo)
 
 If you need another library to develop your parser:
```
Comment (ChucKN0risK) [inline README.md:52]: Replace "If you need another library to develop your parser:" by "Need another library to develop your parser?"

File: README.md [hunk 8]
```diff
@@ -56,7 +59,8 @@ If you need another library to develop your parser:
 To easily create and test your parsers, we advise you to use them on design tokens provided in the `seeds.json` file.
 It will allow you to use our fake tokens to test your parsers.
 
-To use our design tokens seed: 
+To use our design tokens seed:
```
Comment (ChucKN0risK) [inline README.md:62]: Replace "seed" by "seeds"

File: README.md [hunk 9]
```diff
@@ -56,7 +59,8 @@ If you need another library to develop your parser:
...
+
 1. Import it in your [parser].spec.ts using `import * as seeds from '../../seeds.json';`
 2. Use the `seeds.tokens` variable according to your needs.
```
Comment (ChucKN0risK) [inline README.md:65]: Remove the period

File: create-parser.sh [hunk 10]
```diff
@@ -0,0 +1,41 @@
+#/bin/sh
+
+name=$1;
+
+if [ "$name" = "" ]
+then
+  echo "Please name your parser (name needs to be in kebab-case)";
```
Comment (ChucKN0risK) [inline create-parser.sh:7]: Replace "Please name your parser (name needs to be in kebab-case)" by "Please name your parser. It needs to be in kebabcase."

File: create-parser.sh [hunk 11]
```diff
@@ -0,0 +1,41 @@
...
+  exit;
+elif [ -d "./parsers/$name" ]
+then
+  echo "The parser $name already exist. Please choose another name.";
```
Comment (ChucKN0risK) [inline create-parser.sh:11]: Replace "The parser $name already exist. Please choose another name." by "A parser with the same name already exists. Please choose another name."

File: create-parser.sh [hunk 12]
```diff
@@ -0,0 +1,41 @@
...
+rm "parsers/$name/parser-skeleton.parser.ts.bak";
+rm "parsers/$name/parser-skeleton.spec.ts.bak";
+rm "parsers/$name/README.md.bak";
+
+# Rename the files
+mv "./parsers/$name/parser-skeleton.parser.ts" "./parsers/$name/$name.parser.ts";
+mv "./parsers/$name/parser-skeleton.spec.ts" "./parsers/$name/$name.spec.ts";
+
+# Command to test only the created parser
+echo "You can now develop your parser. Go to parsers/$name and start developing 💪"
```
Comment (ChucKN0risK) [inline create-parser.sh:39]: Replace "You can now develop your parser. Go to parsers/$name and start developing 💪" by "You can now create your parser. Go to parsers/$name 💪"

File: create-parser.sh [hunk 13]
```diff
@@ -0,0 +1,41 @@
...
+echo "You can launch your tests using yarn test parsers/$name/$name.spec.ts"
```
Comment (ChucKN0risK) [inline create-parser.sh:40]: Remove "You can"

File: create-parser.sh [hunk 14]
```diff
@@ -0,0 +1,41 @@
...
+echo "You can find more informations on https://specifyapp.com/developers/parsers"
```
Comment (ChucKN0risK) [inline create-parser.sh:41]: Replace "You can find more informations on https://specifyapp.com/developers/parsers" by "Learn more on https://specifyapp.com/developers/parsers".

**Review by ChucKN0risK — CHANGES_REQUESTED** [review]

**Pull request closed without merging**